src/fidesops/service/connectors/query_config.py

```diff
@@ -666,15 +664,47 @@ def generate_query(
         for string_path, reference_values in filtered_data.items():
             for value in reference_values:
                 request_params.append(
-                    self.prepare_params(current_request, {string_path: value})
+                    self.prepare_query_params(current_request, {string_path: value})
                 )
         logger.info(f"Populated request params for {current_request.path}")
         return request_params
 
+    @staticmethod
+    def prepare_update_params(
+        request: SaaSRequest, param_values: Dict[str, Any], body: Dict[str, Any]
+    ) -> SaaSRequestParams:
+        """
+        Populates the placeholders in the request with the given param values
+        """
+        path: str = request.path
+        params: Dict[str, Any] = {}
+
+        for param in request.request_params:
+            if param.type == "query":
+                if param.default_value:
+                    params[param.name] = param.default_value
+                elif param.references:
+                    params[param.name] = pydash.get(
+                        param_values, param.references[0].field
```

Comment:
Does this need to take into account the direction of the reference, whether "from" or "two" has been defined?  What if there are multiple references? I am a little unclear about what it is happening here.

Reply:
Restricting request_param references to only use the `from` direction